src/app/d2-loadout-builder/types.ts

```diff
@@ -1,10 +1,11 @@
 import { D2Item } from '../inventory/item-types';
+import { DestinyInventoryItemDefinition } from 'bungie-api-ts/destiny2';
 
 export type StatTypes = 'STAT_MOBILITY' | 'STAT_RESILIENCE' | 'STAT_RECOVERY';
 
-export interface LockType {
+export interface LockedItemType {
   type: 'item' | 'perk' | 'exclude';
-  items: D2Item[];
+  item: D2Item | DestinyInventoryItemDefinition;
```

Comment:
A fun thing you can do here is to define a function like:

```ts
function isInventoryItemDefinition(item: D2Item | DestinyInventoryItemDefinition): item is DestinyInventoryItemDefinition {
  return (item as DestinyInventoryItemDefinition).displayProperties;
}
```

Then you can use that test like this:

```ts
if (isInventoryItemDefinition(item)) {
  // in here item is a DestinyInventoryItemDefinition
} else {
  // in here it's a D2Item
}
```

Reply:
Added it to that util file, though maybe it would make more sense in there. Typescript amazes me as to how smart it is!